snow/engine/snowman/engine_test.go

```diff
@@ -3182,3 +3184,40 @@ func TestShouldIssueBlock(t *testing.T) {
 		})
 	}
 }
+
+type mockConnVDR struct {
+	tracker.Peers
+	percent float64
+}
+
+func (m *mockConnVDR) ConnectedPercent() float64 {
+	return m.percent
+}
+
+type logBuffer struct {
+	bytes.Buffer
+}
+
+func (logBuffer) Close() error {
+	return nil
+}
```

Comment:
I'm shocked that there is an `io.NopCloser` for an `io.Reader` but not for an `io.Writer`... pain.